src/core/Akka.Streams/Stage/GraphStage.cs

```diff
@@ -423,6 +425,16 @@ public Timer(int id, ICancelable task)
     /// </summary>
     public abstract class GraphStageLogic : IStageLogging
     {
+        private List<Func<object, Task>>? CallbacksWaitingForInterpreter = null;
+        private AtomicReference<List<TaskCompletionSource<Done>>?> AsyncCallbacksInProgress = new(new List<TaskCompletionSource<Done>>());
+        public static readonly TaskCompletionSource<Done> NoPromise;
```

Comment:
Marker for callbacks that doesn't need to observe any task completion (fire and forget)